share/eds/store.go

```diff
@@ -197,16 +200,42 @@ func (s *Store) Blockstore() bstore.Blockstore {
 	return s.bs
 }
 
-// CARBlockstore returns the IPFS blockstore that provides access to the IPLD blocks stored in an
-// individual CAR file.
-func (s *Store) CARBlockstore(ctx context.Context, dataHash []byte) (dagstore.ReadBlockstore, error) {
+// CARBlockstore returns the IPFS blockstore and DAH that provides access to the IPLD blocks stored
+// in an individual CAR file.
+func (s *Store) CARBlockstore(
+	ctx context.Context,
+	dataHash []byte,
+) (dagstore.ReadBlockstore, *header.DataAvailabilityHeader, error) {
```

Comment:
Pls, use `share.Root` instead. Also, it should be `share.DataHash` ,but for some reason it wasn't

Reply:
ah yeah, probably was merged before share.DataHash, good catch

Reply:
fixed